source/collections/LinkedList.ooc

```diff
@@ -117,13 +116,15 @@ LinkedList: class <T> {
 		}
 		result
 	}
-	removeNode: func (toRemove: Node<T>) {
+	removeNode: func (toRemove: Node<T>) -> T {
 		toRemove prev next = toRemove next
 		toRemove next prev = toRemove prev
 		toRemove prev = null
 		toRemove next = null
+		data := toRemove data
 		toRemove free()
 		this _size -= 1
+		data
```

Comment:
Not really like that, but you get the point. It should just take the value away from the Cell object so it does not try to free it.